mozetl/search/daily_search_rollups.py

```diff
@@ -1,309 +1,371 @@
-             r.default_provider or 'NO_DEFAULT')
-        if k in d:
-            searches, people, share = d[k]
-        else:
-            searches, people, share = 0, 0, 0
-        d[k] = (searches + (r.search_count or 0),
-                people + r.profile_count,
-                share + round(r.profile_share, 2))
-    return d
-
-
-def dump_dict(d, basename):
-    """ Write the search roll ups to s3. """
-    bad = []
-    processed = DT.date.today().strftime('%Y-%m-%d')
-    with open(basename, 'w') as f:
-        writer = csv.writer(f)
-        for k, v in d.iteritems():
-            (submission_date, search_provider, country, locale,
-             distribution_id, default_provider) = k
-            search_count, profile_count, profile_share = v
-            try:
-                row = (to_iso(submission_date), search_provider,
-                       str(search_count), country, locale,
-                       distribution_id,
-                       default_provider,
-                       str(profile_count), str(profile_share), processed)
-                row = [s.encode("utf-8") for s in row]
-                writer.writerow(row)
-            except Exception:
-                bad.append((k, v))
-    copy = "aws s3 cp {} {}".format(basename, ROLLUP_WRITE_STEM)
-    subprocess.check_call(copy, shell=True)
-    return bad
-
-
-def write_manifest(date, version, *paths):
-    """ Write a manifest file with the location of the daily rollup files. """
-    text = '\n'.join(paths) + '\n'
-    tries = version + 10
-    while True:
-        manifest_basename = 'daily-search-rollup-manifest-{}-v{}.txt'.format(
-            date, version)
-        path = MANIFEST_WRITE_STEM + manifest_basename
-        if subprocess.call("aws s3 ls {}".format(path), shell=True):
-            break
-        version += 1
-        if version == tries:
-            raise Exception("Can't find unwritten manifest at {}".format(path))
-    with open(manifest_basename, 'w') as f:
-        f.write(text)
-    copy = "aws s3 cp {} {}".format(manifest_basename, MANIFEST_WRITE_STEM)
-    subprocess.check_call(copy, shell=True)
-
-
-owner = 'spenrose'  # XXX read from environment
-owner += '@mozilla.com'
-
-
-def report(date, path, bad):
-    subject = "Daily search rollups for {}".format(date)
-    if bad:
-        subject += ": %d failures" % len(bad)
-    body = 'written to {}'.format(path)
-    if bad:
-        body += '\nproblem rows:\n'
-        for row in bad[:10]:
-            body += str(row) + '\n'
-
-    msg = Message()
-    msg.set_payload(body)
-    msg['Subject'] = subject
-    msg['To'] = owner
-    msg['From'] = owner
-    smtp = smtplib.SMTP('localhost')
-    smtp.sendmail(owner, owner, msg.as_string())
+
+    # persist the dataframe to disk
+    logging.info("Writing dataframe to {}/{}".format(bucket, key))
+    utils.write_csv_to_s3(dataframe.select(select_expr), bucket, key, header=False)
+
+    csv_paths = get_csv_locations(bucket, key)
```

Comment:
Nit: I would be inclined to define the paths we will write in one place, rather than relying on retrieving them later ... particularly from a filesystem.